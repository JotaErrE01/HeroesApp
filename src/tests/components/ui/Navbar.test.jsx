import { render, screen } from "@testing-library/react"
import userEvent from "@testing-library/user-event";
import {  Router } from "react-router";
import AuthContext from "../../../auth/AuthContext";
import { Navbar } from "../../../components/ui/Navbar";
import { createMemoryHistory } from 'history';


describe('Pruebas en el <Navbar />', () => {

    const history = createMemoryHistory();

    const contextValue = {
        dispatch: jest.fn(), // siempre que quieras probar funciones
        user: {
            logged: true,
            name: 'Jonathan'
        }
    }

    let wrapper;

    beforeEach(() => {
        wrapper = render(
            <AuthContext.Provider value={contextValue}>
                <Router history={history}>
                    <Navbar />
                </Router>
            </AuthContext.Provider>
        );
    });
    
    test('Debe de mostrarse correctamente', () => {
       
        expect( wrapper.container ).toMatchSnapshot();
        expect( wrapper.container.querySelector('.text-info').textContent.trim() ).toBe('Jonathan');

    });

    test('Debe de llamar el logOut y usar history', () => {
       
        userEvent.click( screen.getByText('Logout') );

        expect( contextValue.dispatch ).toHaveBeenCalled();
        expect( history.location.pathname ).toBe( '/login' );

    });

})