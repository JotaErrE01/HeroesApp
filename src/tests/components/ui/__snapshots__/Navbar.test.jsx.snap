// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el <Navbar /> Debe de mostrarse correctamente 1`] = `
<div>
  <nav
    class="navbar navbar-expand-sm navbar-dark bg-dark"
  >
    <a
      class="navbar-brand"
      href="/"
    >
      Asociaciones
    </a>
    <div
      class="navbar-collapse"
    >
      <div
        class="navbar-nav"
      >
        <a
          class="nav-item nav-link"
          href="/marvel"
        >
          Marvel
        </a>
        <a
          class="nav-item nav-link"
          href="/dc"
        >
          DC
        </a>
        <a
          class="nav-item nav-link"
          href="/search"
        >
          Search
        </a>
      </div>
    </div>
    <div
      class="navbar-collapse collapse w-100 order-3 dual-collapse2"
    >
      <ul
        class="navbar-nav ml-auto"
      >
        <span
          class="nav-item nav-link text-info"
        >
          Jonathan
        </span>
        <button
          class="nav-item nav-link btn"
        >
          Logout
        </button>
      </ul>
    </div>
  </nav>
</div>
`;
